src/Ocelot/Requester/MessageInvokerPool.cs

```diff
@@ -0,0 +1,108 @@
+﻿using Ocelot.Configuration;
+using Ocelot.Logging;
+using System.Net.Security;
+
+namespace Ocelot.Requester;
+
+public class MessageInvokerPool : IMessageInvokerPool
+{
+    // TODO This should be configurable and available as global config parameter in ocelot.json
+    public const int DefaultRequestTimeoutSeconds = 90;
+
+    private readonly ConcurrentDictionary<MessageInvokerCacheKey, Lazy<HttpMessageInvoker>> _handlersPool;
+    private readonly IDelegatingHandlerHandlerFactory _handlerFactory;
+    private readonly IOcelotLogger _logger;
+
+    public MessageInvokerPool(IDelegatingHandlerHandlerFactory handlerFactory, IOcelotLoggerFactory loggerFactory)
+    {
+        _handlerFactory = handlerFactory ?? throw new ArgumentNullException(nameof(handlerFactory));
```

Comment:
Maybe default factory instance, very simple and lite, is better than exception?

Reply:
What about this issue?
Does it make sense to have a default factory and removing the exception finally?

Reply:
Excuse me, is `IDelegatingHandlerHandlerFactory` interface registered in DI?

Reply:
Yes, as a singleton, the MessageInvokerPool too... By the way, we should have a look at the services lifetime. I can only see singletons, it's a bit odd. But I think we already talked about this.

Reply:
Ok, not an issue! Redundant argument null-check is not a problem.
Seems this Pooler service should be shared service aka singleton in DI.
Why do you want review service lifetime?
Does Pooler use shared (static) block of memory? I'm lazy to look into the code...

Reply:
No fine, my bad... But still, I'm a bit surprised, we have Singeltons all over the place.